### PR TITLE
Upload a transcoded video to youtube

### LINKED ISSUE #480: Upload transcoded videos to Youtube
As a user, I'd like videos transcoded from Google Drive also uploaded to Youtube

#### Designs and Mockups
- The existing OVS code for doing this will be re-used to the extent possible.

#### Acceptance Criteria:

- [ ] Videos are uploaded to Youtube after transcoding.

### PR DESCRIPTION
**Pull request by mbertrand**
Depends on #486 , but you should be able to test this by creating your own Youtube API project as described in the new README section.

#### Pre-Flight checklist

- [x] Migrations
  - [x] Migration is backwards-compatible with current production code
- [x] Testing
  - [x] Code is tested
  - [x] Changes have been manually tested
- [x] Settings
  - [x] New settings are documented and present in `app.json`
  - [x] New settings have reasonable development defaults, if applicable
  - [x] Opened issue for DevOps regarding necessary configuration changes to deployed environments

#### What are the relevant tickets?
Closes #480

#### What's this PR do?
When AWS MediaConvert transcoding is complete, upload the appropriate video file to Youtube.

#### How should this be manually tested?
- Create your own youtube api configuration following the steps in the new README section.
- Set up Google Drive integration and transcoding as described for PR #469.  You will need to simulate the webhook endpoint request so that the required`VideoFile` objects will be created and the youtube upload task triggered.  The request body should contain this (update to match your settings/video/drive file):
   ```
   {
	"version": "0",
	"id": "426c759f-277e-ee0f-aceb-7fb79214bc51",
	"detail-type": "MediaConvert Job State Change",
	"source": "aws.mediaconvert",
	"account": "<settings.AWS_ACCOUNT_ID>"
	",
	"detail": {
		"timestamp": 1629911639065,
		"accountId": "919801701561",
		"queue": "arn:aws:mediaconvert:us-east-1:919801701561:queues/Default",
		"jobId": "<VideoJob.job_id>",
		"status": "COMPLETE",
		"userMetadata": {},
		"outputGroupDetails": [{
			"outputDetails": [{
					"outputFilePaths": [
						"s3://<settings.AWS_STORAGE_BUCKET_NAME>/aws_mediaconvert_transcodes/<Website.short_id>/<DriveFile.file_id>/<original_video_filename_base>_youtube.mp4"
					],
					"durationInMs": 45466,
					"videoDetails": {
						"widthInPx": 320,
						"heightInPx": 176
					}
				},
				{
					"outputFilePaths": [
						"s3://<settings.AWS_STORAGE_BUCKET_NAME>/aws_mediaconvert_transcodes/<Website.short_id>/<DriveFile.file_id>/<original_video_filename_base>_360p_16_9.mp4"
					],
					"durationInMs": 45466,
					"videoDetails": {
						"widthInPx": 640,
						"heightInPx": 360
					}
				},
				{
					"outputFilePaths": [
						"s3://<settings.AWS_STORAGE_BUCKET_NAME>/aws_mediaconvert_transcodes/<Website.short_id>/<DriveFile.file_id>/<original_video_filename_base>_360p_4_3.mp4"
					],
					"durationInMs": 45466,
					"videoDetails": {
						"widthInPx": 480,
						"heightInPx": 360
					}
				}
			],
			"type": "FILE_GROUP"
		}]
	}
  }
  ```
- Wait a few minutes, the `upload_youtube_videos` and `update_youtube_statuses` tasks should run.  Go to ../admin/videos/video/<video_id>/change/  and check that the `VideoFile` with `destination` of `youtube` now has a `destination_id` and a desination_status of either `uploaded` (initially) or `processed` (ultimately).
Once the `destination_status` is `processed`, the `status` should be `Complete`.
- Go to `https://www.youtube.com/watch?v=<destination_id>`, your video should play from there.

#### Any background context you want to provide?
Much of this code was originally from OVS and modified as necessary.

### CODE REVIEW

File: videos/tasks.py [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+        except:  # pylint: disable=bare-except
+            log.exception("Error uploading video to Youtube: %s", video_file.s3_key)
+            video_file.status = VideoFileStatus.FAILED
+        video_file.save()
+
+
+@app.task
+def update_youtube_statuses():
+    """
+    Update the status of recently uploaded YouTube videos and upload captions if complete
```
Comment (abeglova) [inline videos/tasks.py:63]: I don't think this function is uploading captions

File: videos/youtube.py [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+        Returns:
+            dict: List of captions in JSON format
+        """
+        results = (
+            self.client.captions().list(part="snippet", videoId=video_id).execute()
+        )
+
+        return {item["snippet"]["language"]: item["id"] for item in results["items"]}
+
+    def upload_caption(self, caption, video_id):
```
Comment (abeglova) [inline videos/youtube.py:146]: Does this come from the ol-video-services codebase? Doesn't look like it's currently used in this PR but makes sense to upload to youtube when we get a new caption file from 3play

**Review by abeglova — APPROVED** [review]
👍

File: videos/youtube.py (hunk 2 above)
Comment (mbertrand) [reply videos/youtube.py:146]: Thanks for catching this, the functions were copied from OVS but I think it's likely they will need to be tweaked once 3play captions are available, so I'm going to cut them out, then add them back in a future PR once they're needed.

**Review by abeglova — APPROVED** [review]
LGTM once the tests pass

**Comment by mbertrand** [comment]
Waiting to merge until the CI/RC youtube API quota gets raised above 0.

**Pull request merged**